### PR TITLE
[DTensor][Test] Add unit tests to keep track of DTensor sharding for 2D

### PR DESCRIPTION
**Pull request by wz337**
Fixes #ISSUE_NUMBER

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: test/distributed/_tensor/test_utils.py [hunk 1]
```diff
@@ -119,5 +123,82 @@ def test_compute_local_shape_and_global_offset_2D(self):
                 )
 
 
+class TestFSDP2DLocalShardsAndOffsets(DTensorTestBase):
```
Comment (wanchaol) [inline test/distributed/_tensor/test_utils.py:126]: I think these tests should be in FSDP2 test folder? Or maybe rename the test cases to something like strided_local_shard tests and we can probably leave it in DTensor

**Review by wanchaol — COMMENTED** [review]
Thanks for adding the tests! One comment

File: test/distributed/_tensor/test_utils.py [hunk 2]
```diff
@@ -119,5 +123,82 @@ def test_compute_local_shape_and_global_offset_2D(self):
...
+        return 4
+
+    @with_comms
+    def test_fsdp1_tp_2d_dtensor_local_shards_and_offsets(self):
+        # We are mimicking the behavior of FSDP1 + TP.
+        # Currently, the 2D DTensor's local shard is correct, since from_local + redistribute incurs a all_gather behind the scene.
+        # When we have a global_tensor of [0, 1, 2, 3, 4, 5, 6, 7], the local shard of 2D DTensor would be:
+        # rank0: [0, 1], rank1: [2, 3], rank2: [4, 5], rank3: [6, 7]
+        comm_mode = CommDebugMode()
+        with comm_mode:
```
Comment (awgu) [inline test/distributed/_tensor/test_utils.py:138]: nit: Can this be `with CommDebugMode() as comm_mode:`?

Comment (wz337) [reply test/distributed/_tensor/test_utils.py:138]: Good catch. Definitely was not thinking lol.

Comment (wz337) [reply test/distributed/_tensor/test_utils.py:138]: Oops realized I do actually need `comm_mode` for checking the number of collective call later on. Changing it back.

Comment (awgu) [reply test/distributed/_tensor/test_utils.py:138]: If we do `with CommDebugMode() as comm_mode`, I thought we can still access `comm_mode`.

**Review by wanchaol — APPROVED** [review]

**Pull request closed without merging**